api/handlers.go

```diff
@@ -55,7 +59,79 @@ func (s *Server) Pulse(ctx echo.Context, pulseNumber server.PulseNumberPathParam
 }
 
 func (s *Server) JetDropsByPulseNumber(ctx echo.Context, pulseNumber server.PulseNumberPathParam, params server.JetDropsByPulseNumberParams) error {
-	panic("implement me")
+	if !pulse.IsValidAsPulseNumber(int(pulseNumber)) {
+		response := server.CodeValidationError{
+			Code:               NullableString("400"),
+			Description:        NullableString("Invalid pulse number"),
+			Link:               nil,
+			Message:            nil,
+			ValidationFailures: nil,
+		}
+		return ctx.JSON(http.StatusBadRequest, response)
+	}
+
+	limit, offset, err := checkLimitOffset(params.Limit, params.Offset)
+	if err != nil {
+		response := server.CodeValidationError{
+			Code:               NullableString("400"),
+			Description:        NullableString("Invalid limit or offset query parameters"),
+			Link:               nil,
+			Message:            NullableString(err.Error()),
+			ValidationFailures: nil,
+		}
+		return ctx.JSON(http.StatusBadRequest, response)
+	}
+
+	jetDropID, err := checkJetDropID(params.FromJetDropId)
+	if err != nil {
+		response := server.CodeValidationError{
+			Code:               NullableString("400"),
+			Description:        NullableString("Invalid jet drop id in query parameters"),
+			Link:               nil,
+			Message:            NullableString(err.Error()),
+			ValidationFailures: nil,
+		}
+		return ctx.JSON(http.StatusBadRequest, response)
+	}
+	jetDrops, err := s.storage.GetJetDrops(
+		models.Pulse{PulseNumber: int(pulseNumber)},
+		jetDropID,
+		&limit,
+		&offset,
+	)
+	if err != nil {
+		s.logger.Error(err)
+		return ctx.JSON(http.StatusInternalServerError, struct{}{})
+	}
+
+	var result []server.JetDrop
+	for _, jetDrop := range jetDrops {
+		result = append(result, JetDropToAPI(jetDrop))
```

Comment:
actually it multiplies len it on 2, faster, but don't know if it really matters

Reply:
Actually it is a slice, it have length and cap, cap is a free space in slice.
But here is minor question, can we use []*server.JetDrop to minimise slice allocations ? or it will be more expensive due to garbage collecting of constructed jetdrops ?